packit/utils/obs_helper.py

```diff
@@ -0,0 +1,454 @@
+            break
+
+        if file.endswith(".changes"):
+            logger.info(".changes file already exists in package")
+            return
+
+    if not specfile:
+        raise ValueError("Cannot find spec file in package")
+
+    with open(specfile) as s_file:
+        content = s_file.read()
+
+    split_content = re.split(r"%changelog", content, flags=re.IGNORECASE)
+    if len(split_content) < 2:
+        logger.info("Project has no changelog")
+        return
+
+    specfile_content, changes_file_content = split_content[0], "".join(
+        split_content[1:],
+    )
+
+    with open(package_dir / specfile, "w") as s_file:
+        specfile_content = "\n".join(
+            [
+                release_line if line.lower().startswith("release:") else line
+                for line in specfile_content.splitlines()
+            ],
+        )
+        s_file.write(specfile_content)
+
+    changes_filename = package_dir / f"{specfile[:-5]}.changes"
+
+    changes_file_content = format_changelog(changes_file_content)
+
+    try:
+        with open(changes_filename, "w") as changes_file:
+            changes_file.write(changes_file_content)
+    except OSError as err:
+        logger.warn(f"Error writing {changes_filename} file: {err}")
+
+
+def create_package(project_name: str, package_name: str) -> None:
+    """Creates the package with the name ``package_name`` in the project
+    ``project_name`` on OBS. No sources are uploaded in the process.
+
+    """
+    (root := ET.Element("package")).attrib["name"] = package_name
+    root.attrib["project"] = project_name
+
+    (title := ET.Element("title")).text = f"The {package_name} package"
+    descr = ET.Element("description")
+
+    root.append(title)
+    root.append(descr)
+
+    package_url = core.makeurl(
+        _API_URL,
+        ["source", project_name, package_name, "_meta"],
+    )
+    metafile = core.metafile(package_url, ET.tostring(root))
+    metafile.sync()
+
+
+def create_obs_project(
+    project: str,
+    targets: str,
+    owner: Optional[str],
+    package_config: PackageConfig,
+    description: Optional[str],
+):
+    """
+    Creates a new OBS project and its associated package, ensuring only a single
+    package is included per call.
+
+    Args:
+        project (str, optional): The desired name for the OBS project. Defaults to None.
+        targets (str): Comma-separated list of build targets for the project. Defaults to "".
+        owner (Optional[str], optional): The owner of the project. Defaults to None.
+        package_config (PackageConfig): The config for the package to be included in the project.
+        description (Optional[str], optional): A description for the project. Defaults to None.
+
+    Returns:
+        Tuple[str, str]: A tuple containing the created project name and associated package name.
+
+    Raises:
+        ValueError: If the provided package_config contains multiple packages.
+    """
+    conf.get_config()
```

Comment:
The config is being loaded into the module itself as global?

Reply:
It's a global. osc supports just one config per user.